src/backend/access/aocs/aocsam.c

```diff
@@ -1178,12 +1180,22 @@ aocs_gettuple(AOCSScanDesc scan, int64 targrow, TupleTableSlot *slot)
 		DatumStreamRead *ds = scan->columnScanInfo.ds[attno];
 		int64 startrow = scan->segfirstrow + scan->segrowsprocessed;
 		bool chkvisimap = true;
+		bool isanchor = i == ANCHOR_COL_IN_PROJ;
+
+		/*
+		 * We should only use targrow for the initial anchor column scan.
+		 * phyrow should be used for all other columns, since it uniquely
+		 * identifies the row in all column files, unlike the sequential
+		 * number that can be different for different columns.
+		 */
+		int64 current_target_row = isanchor ? targrow : phyrow;
+		int64 current_start_row;
```

Comment:
can we initialize this value with something here?

Reply:
We can with `isanchor ? startrow : ds->blockFirstRowNum + ds->blockRowsProcessed;`, but it's possibly not a valid value (because `ds` wasn't updated) and it will be overwritten later anyway. Do we need it? I can also just zero-initialize it here.

Reply:
`InvalidAORowNum` ?

Reply:
Yes, it was suggested to move it there since all the other variables are declared there too. Should I use `InvalidAORowNum` or should I move it to where it is used?

Reply:
If it was desided to declare everything at one place, probably it would be enough to just init via `InvalidAORowNum`